### PR TITLE
Separate database connections for voyager and other tables to BREAD

### PR DESCRIPTION
**Pull request by fbuccioni**
Hi, I'm trying to use Voyager as admin console of a database and I saw the issue #69 trying to use the same feature.

I made a different base class of `Model` and `User` that gets the `$connection` property from `voyager.database.connection` config.

Steps to use it:

1. Configure both db, voyager and main in `config/database.php`
2. Set voyager db connection as default in `config/database.php`
3. Install voyager `artisan voyager:install` 
4. Set `voyager.database.connection` variable as voyager db connection in `config/voyager.php`
5. Set main db connection as default in `config/database.php`

### CODE REVIEW

File: src/Traits/Authenticatable.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace TCG\Voyager\Traits;
+
+use Illuminate\Foundation\Auth\User;
+
+class Authenticatable extends User
```
Comment (fletch3555) [inline src/Traits/Authenticatable.php]: This is a class, not a trait, and shouldn't be a trait.  Please change the namespace.

File: src/VoyagerServiceProvider.php [hunk 2]
```diff
@@ -270,22 +269,20 @@ public function registerGates()
         // otherwise it will throw an error because no database
         // connection has been made yet.
         try {
-            if (Schema::hasTable('data_types')) {
```
Comment (fletch3555) [inline src/VoyagerServiceProvider.php:273]: What's the reasoning behind removing this check?

File: src/Models/User.php [hunk 3]
```diff
@@ -3,8 +3,8 @@
 namespace TCG\Voyager\Models;
 
 use Carbon\Carbon;
-use Illuminate\Foundation\Auth\User as Authenticatable;
 use TCG\Voyager\Contracts\User as UserContract;
+use TCG\Voyager\Traits\Authenticatable;
```
Comment (fletch3555) [inline src/Models/User.php]: Same comment, Authenticatable is a class, not a trait.

**Review by fletch3555 — CHANGES_REQUESTED** [review]
I don't really agree with these changes.  Models will already use the configured "default" connection.  If you need this to be different, just change `$this->connection` on your model class.

Please explain your use-case a bit more and I may reconsider

File: src/VoyagerServiceProvider.php (hunk 2 above)
Comment (fbuccioni) [reply src/VoyagerServiceProvider.php:273]: Just in the debug stage I remove that if, now are ok

File: src/Traits/Authenticatable.php (hunk 1 above)
Comment (fbuccioni) [reply src/Traits/Authenticatable.php]: Done

File: src/Models/User.php (hunk 3 above)
Comment (fbuccioni) [reply src/Models/User.php]: Done

**Comment by falcacibar** [comment]
My use case is because I have a db with several tables in another location and I'd like to not use that database to store the voyager tables, keeping it organized.

I need to make a mantainer and admin interface for that database so I use voyager because it fits right in the profile I want to use.

When I use separated connection in models I cannot use the database tool and I don't have the tool to create or edit new BREADs.

**Comment by fbuccioni** [comment]
My use case is because I have a db with several tables in another location and I'd like to not use that database to store the voyager tables, keeping it organized.

I need to make a mantainer and admin interface for that database so I use voyager because it fits right in the profile I want to use.

When I use separated connection in models I cannot use the database tool and I don't have the tool to create or edit new BREADs.

**Pull request closed without merging**